src/function/cast_string_non_nested_functions.cpp

```diff
@@ -83,5 +88,94 @@ bool TryCastStringToTimestamp::tryCast<timestamp_sec_t>(const char* input, uint6
     return true;
 }
 
+static bool isDate(const std::string& str) {
+    return RE2::FullMatch(str, "\\d{4}/\\d{1,2}/\\d{1,2}") ||
```

Comment:
I think the first match is trying to match date with style: 
2023/8/7, 1999/12/31.
I don't think our csv reader support such format.

Reply:
We support all three formats. I have a test for each in `date.csv`

Reply:
I think we support those separators:
```
    if (sep != ' ' && sep != '-' && sep != '/' && sep != '\\') {
        // invalid separator
        return false;
    }
```
Can you add those separators there?